components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java

```diff
@@ -8657,6 +8672,58 @@ public boolean isFavouriteApp(APIIdentifier identifier, String username, int ten
         return apiIdentifiers;
     }
 
+    private void getUserAccessibleAppsByBusinessOwner(List<APIIdentifier> apiIdentifiers, Map<String, List<String>>
+            businessOwnerIdsMap, Registry registry, int tenantId, String userName) throws AppManagementException {
+        boolean isTenantFlowStarted = false;
+        try {
+            UserRealmService realmService =
+                    (UserRealmService) PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                            .getOSGiService(UserRealmService.class);
+            String tenantDomain = realmService.getTenantManager().getDomain(tenantId);
+
+            if (tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
```

Comment:
Rename 'tenantDomain' as 'requestedTenantDomain'. Need to highlight logged-in user and store belongs to two tenants. Otherwise it's not clear why we setting tenantdomain in caronContext.